builders/nodejs.go

```diff
@@ -160,7 +161,48 @@ func (builder *NodeJSBuilder) IsModule(target string) (bool, error) {
 	return false, errors.New("IsModule is not implemented for NodeJSBuilder")
 }
 
-// InferModule is not implemented
-func (builder *NodeJSBuilder) InferModule(target string) (module.Module, error) {
-	return module.Module{}, errors.New("InferModule is not implemented for NodeJSBuilder")
+// DiscoverModules builds ModuleConfigs for any package.jsons that are not contained in a node_modules dir
+func (builder *NodeJSBuilder) DiscoverModules(dir string) ([]config.ModuleConfig, error) {
+	moduleConfigs := []config.ModuleConfig{}
```

Comment:
You should use the `var slice []T` form unless the length is known, in which case use `make`.